utils/build-script-impl

```diff
@@ -1240,6 +1240,18 @@ if [[ ! "${SKIP_BUILD_SWIFTPM}" ]] ; then
      PRODUCTS=("${PRODUCTS[@]}" swiftpm)
 fi
 
+# Checks if a given product is enabled (i.e. part of $PRODUCS array)
+function contains_product() {
```

Comment:
nit-pick: Since this is only used in one spot below, I personally wouldn't split it out into its own function. Your call, though!